### PR TITLE
getAllPossiblePropertiesOfTypes: Skip primitives

### LINKED ISSUE #15894: filter primitives in `getAllPossiblePropertiesOfType`
```ts
declare function foo(arg: string | { x: number });
foo({ | });
```

This shows all the members of `string`, which is not intended here. i would say we need to filter the primitives from these list.

image

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #15894

### CODE REVIEW

File: tests/cases/fourslash/completionListOfUnion.ts [hunk 1]
```diff
@@ -10,9 +10,11 @@
 ////f({ /*f*/ });
 
 goTo.marker("x");
+verify.completionListCount(3);
 verify.completionListContains("a", "(property) a: string | number");
 verify.completionListContains("b", "(property) b: number | boolean");
 verify.completionListContains("c", "(property) c: string");
 
 goTo.marker("f");
 verify.completionListContains("a", "(property) a: number");
```
Comment (mhegazy) [inline tests/cases/fourslash/completionListOfUnion.ts]: worth adding something with `T | null` and `T|undefined` to make sure we do not break these in the future.

Comment (andy-ms) [inline tests/cases/fourslash/completionListOfUnion.ts]: You mean:
```ts
function g<T>(x: T | null) {}
g({ /*completion*/ });
```

Comment (mhegazy) [inline tests/cases/fourslash/completionListOfUnion.ts]: no i meant:
```ts
function g(x: {a: number} | null) {}
g({   | /* should see "a" */ })
```

Comment (andy-ms) [inline tests/cases/fourslash/completionListOfUnion.ts]: Should be handled by the first test then. Added `| undefined` too.

Comment (mhegazy) [inline tests/cases/fourslash/completionListOfUnion.ts]: sorry did not see that earlier. thanks!

**Pull request merged**